prow/apis/prowjobs/v1/types.go

```diff
@@ -115,6 +115,8 @@ type ProwJobSpec struct {
 	// ExtraRefs are auxiliary repositories that
 	// need to be cloned, determined from config
 	ExtraRefs []Refs `json:"extra_refs,omitempty"`
+	// JenkinsJob holds configuration specific to Jenkins jobs
+	JenkinsJob *JenkinsJob `json:"jenkins_job,omitempty"`
```

Comment:
nit: could you call this `JenkinsSpec` and put it next to the `PodSpec` and `BuildSpec` below?